### PR TITLE
For #25736: Add experiment for Total cookie protection.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactory.kt [hunk 1]
```diff
@@ -106,10 +107,16 @@ class TrackingProtectionPolicyFactory(
 }
 
 @VisibleForTesting
-internal fun TrackingProtectionPolicyForSessionTypes.adaptPolicyToChannel(): TrackingProtectionPolicyForSessionTypes {
+internal fun TrackingProtectionPolicy.applyTCPIfNeeded(settings: Settings): TrackingProtectionPolicyForSessionTypes {
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactory.kt:110]: Repurposing `adaptPolicyToChannel`, as it wasn't use anymore see https://github.com/mozilla-mobile/fenix/pull/23056

File: nimbus.fml.yaml [hunk 2]
```diff
@@ -202,6 +202,17 @@ features:
...
+    description: Enable the total cookie protection to some elegible users.
+    variables:
+      enabled:
+        description: If true, the total cookie protection in all browsing modes.
+        type: Boolean
+        default: false
+    defaults:
+      - channel: developer
+        value:
+          enabled: true
```
Comment (jhugman) [inline nimbus.fml.yaml:215]: I think this feature is a bit small.

Would suggest a feature called `web-view-settings` or `web-view`, and have one of the variables called `total-cookie-protection`.

A good test/rule of thumb for what a feature is would be asking "when do you send an exposure event?", or "does the user know when they have been shown the feature?".

**Review by jhugman — CHANGES_REQUESTED** [review]
I can't speak to the implementation, but I have left some suggestions on how to arrange your Nimbus features.

File: nimbus.fml.yaml (hunk 2 above)
Comment (Amejia481) [reply nimbus.fml.yaml:215]: Thanks for the feedback, I would rather prefer to  `engine-settings` as it's very specific about `WebView` 😅 and we mostly support `GeckoView`.

Comment (Amejia481) [reply nimbus.fml.yaml]: Yes, I added on https://github.com/mozilla-mobile/fenix/pull/25791 because of your feedback :)

**Comment by Amejia481** [comment]
I updated the PR with the suggested changes

File: nimbus.fml.yaml [hunk 3]
```diff
@@ -210,7 +210,10 @@ features:
         description: If true, enables tabs prioritization feature.
         type: Boolean
         default: true
-
+      total-cookie-protection-enabled:
+        description: If true, enables the total cookie protection in all browsing modes.
+        type: Boolean
+        default: false
```
Comment (jhugman) [inline nimbus.fml.yaml:216]: 😻

**Review by jhugman — APPROVED** [review]
Lovely. Thank you for making the changes. 🚢 :shipit:

**Comment by Amejia481** [comment]
@Mergifyio backport releases_v103.0.0

**Pull request merged**